daprdocs/content/en/operations/extending/developing-grpc-components.md

```diff
@@ -0,0 +1,339 @@
+---
+type: docs
+title: "Developing gRPC-based Pluggable Components"
+linkTitle: "Developing gRPC-based Pluggable Components"
+weight: 250
+description: "Extending dapr with external gRPC-based components"
+---
+
+[gRPC-based](https://grpc.io/) Dapr components are typically run as containers or processes that communicate with the dapr main process via [Unix Domain Sockets](https://en.wikipedia.org/wiki/Unix_domain_socket).
+
+## Developing a Pluggable Component from scratch
+
+{{< tabs ".NET" >}}
+{{% codetab %}}
+
+### Step 1: Prerequisites
+
+For this tutorial, you must have minimal knowledge of [gRPC and protocol buffers](https://grpc.io/), and that you chose a programming language [that supports gRPC](https://grpc.io/docs/languages/).
+
+For simplicity, all code samples uses the generated code from the [protoc](https://developers.google.com/protocol-buffers/docs/downloads) tool or supported gRPC build tools by languages.
+
+As previously mentioned, dapr uses a Unix Domain Socket to communicate with gRPC-based components, which means that as a prerequisite you also need a UNIX-like system, or for Windows users, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) should be sufficient.
```

Comment:
I think we could move this particular paragraph to a language-agnostic section and provide links for .Net specific stuff later. What you think?

Reply:
Not sure, [I've found other documentation doing the same](https://github.com/dapr/docs/blob/v1.8/daprdocs/content/en/getting-started/quickstarts/serviceinvocation-quickstart.md), and I think it worth to pay for duplicated text against having two prerequisite sections?